app/controllers/documents_controller.rb

```diff
@@ -4,11 +4,28 @@
 class DocumentsController < ApplicationController
   def index
     @document = Document.new
```

Comment:
There are common query setup for each variable values that you can dry up.
e.g.
```ruby
base_relation  = Document.with_attached_file
base_relation = base_relation.where(published: true) if member_signed_in?
...
```